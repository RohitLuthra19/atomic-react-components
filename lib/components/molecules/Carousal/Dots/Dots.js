// @flow
/**
 *
 * Dots
 *
 */
import * as React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import Slide from '../Slide';
import List from '../../List';
import styles from './Dots.style';
import Button from '../../../atoms/Button';

type Props = {
  currentSlideIndex?: number,
  totalSlides: number,
  children: React.ChildrenArray<React.Element<typeof Slide>>,
  showThumbnails: boolean,
  className?: string,
  onDotClick: Function,
};

const createDotsArray = (currentSlideIndex, totalSlides) =>
  Array.from(new Array(totalSlides), (value, index) => ({
    index,
    isActive: currentSlideIndex === index,
  }));

const Dots = ({
  currentSlideIndex,
  totalSlides,
  children,
  showThumbnails,
  className,
  onDotClick,
}: Props): Node => {
  const dotsArray = createDotsArray(currentSlideIndex, totalSlides);
  return (
    <div className={className}>
      {showThumbnails ? (
        <List className="thumbnail" noDefaultView>
          {React.Children.map(children, (current, index) => (
            <Slide
              key={`slide_${index}`}
              isActive={currentSlideIndex === index}
              onThumbnailClick={() => onDotClick(index)}
            >
              {current}
            </Slide>
          ))}
        </List>
      ) : (
        dotsArray.map(dot => (
          <Button
            key={`dot_${dot.index}`}
            className="dot"
            primary={dot.isActive}
            secondary={!dot.isActive}
            onClick={() => onDotClick(dot.index)}
          />
        ))
      )}
    </div>
  );
};

Dots.defaultProps = {
  currentSlideIndex: 0,
  className: '',
};

export default styled(Dots)`
  ${styles};
`;

export { Dots as DotsVanilla };
