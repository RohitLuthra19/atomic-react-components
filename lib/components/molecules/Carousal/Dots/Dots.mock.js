const defaultConfig = {
  totalSlides: 5,
};

const configWithDefaultIndex = {
  totalSlides: 5,
  currentSlideIndex: 2,
  onDotClick: index => {
    console.log('current index:', index);
  },
};

export { configWithDefaultIndex };
export default defaultConfig;
