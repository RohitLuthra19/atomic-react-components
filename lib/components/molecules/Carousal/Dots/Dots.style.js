import { css } from 'styled-components';

export default css`
  text-align: center;
  .dot {
    width: 10px;
    height: 10px;
    min-height: 10px;
    border-radius: 50%;
    border: 1px solid;
    margin-right: 10px;
    padding: 0;
  }
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;
