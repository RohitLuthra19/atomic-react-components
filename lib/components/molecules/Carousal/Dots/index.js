export { default } from './Dots';
export { DotsVanilla } from './Dots';
