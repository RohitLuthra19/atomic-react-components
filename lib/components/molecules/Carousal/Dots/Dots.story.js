import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import defaultConfig, { configWithDefaultIndex } from './Dots.mock';

// Import Styled Component to showcase variations
import Dots, { DotsVanilla } from '.';

storiesOf('Molecules/Carousal', module).addWithChapters('Dots', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <DotsVanilla
              {...defaultConfig}
              className={`hide-default-sample ${defaultConfig.className}`}
            />
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Primary Dots Variations',
      sections: [
        {
          title: 'Simple Dots',
          sectionFn: () => <Dots {...configWithDefaultIndex} />,
        },
      ],
    },
  ],
});
