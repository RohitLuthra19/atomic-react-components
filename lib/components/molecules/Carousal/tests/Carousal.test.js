import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';

import { CarousalVanilla as Carousal } from '../index';
import Dots from '../Dots';

describe('<Carousal />', () => {
  let CarousalComponent = '';
  beforeEach(() => {
    CarousalComponent = shallow(
      <Carousal showDots>
        <p>Slide 1</p>
        <p>Slide 2</p>
        <p>Slide 3</p>
      </Carousal>
    );
  });

  test('should render correctly', () => {
    expect(CarousalComponent).toMatchSnapshot();
  });
});

describe('Carousal Component', () => {
  let CarousalComponent = '';
  beforeEach(() => {
    CarousalComponent = shallow(
      <Carousal showDots>
        <p>Slide 1</p>
        <p>Slide 2</p>
        <p>Slide 3</p>
      </Carousal>
    );
  });

  test('should make first slide as active on load', () => {
    expect(CarousalComponent.state('current')).toBe(0);
  });

  test('should render left arrow as disabled', () => {
    expect(
      CarousalComponent.find('.arrow')
        .first()
        .prop('disabled')
    ).toBe(true);
  });

  test('on click on right arrow, should increase current state by 1', () => {
    const currentState = CarousalComponent.state('current');
    CarousalComponent.find('.arrow.right').simulate('click');
    expect(CarousalComponent.state('current')).toBe(currentState + 1);
  });

  test('on click on left arrow, should decrease current state  by 1', () => {
    CarousalComponent.setState({ current: 2 });
    CarousalComponent.find('.arrow')
      .first()
      .simulate('click');
    expect(CarousalComponent.state('current')).toBe(1);
  });

  test('should not render arrows if showArrow prop is false', () => {
    CarousalComponent.setProps({
      showArrows: false,
    });
    expect(CarousalComponent.find('.arrow').length).toBe(0);
  });

  test('should not render arrows if showDots prop is false', () => {
    CarousalComponent.setProps({
      showDots: false,
    });
    expect(CarousalComponent.find(Dots).length).toBe(0);
  });
});
