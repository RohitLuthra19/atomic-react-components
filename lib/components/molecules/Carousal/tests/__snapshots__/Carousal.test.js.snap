// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousal /> should render correctly 1`] = `
<section
  className=""
>
  <div
    className="slider-container"
  >
    <Button
      className="arrow"
      disabled={true}
      onClick={[Function]}
    >
      &lt;
    </Button>
    <Slider
      currentSlideIndex={0}
      totalSlides={3}
    >
      <p>
        Slide 1
      </p>
      <p>
        Slide 2
      </p>
      <p>
        Slide 3
      </p>
    </Slider>
    <Button
      className="arrow right"
      disabled={false}
      onClick={[Function]}
    >
      &gt;
    </Button>
  </div>
  <Dots
    currentSlideIndex={0}
    onDotClick={[Function]}
    totalSlides={3}
  />
</section>
`;
